### DISCUSSION TITLE
What does "hubs secret repo" mean in the reticulum readme when trying to run reticulum locally?

### DISCUSSION THREAD

**Discussion by Tsuyoshina** [mozilla/hubs#3323] [General Questions & Discussion] [answered]
Hi,

I'm going through the steps in "https://github.com/mozilla/reticulum" and trying to set it up.
But I'm not sure "hubs secret repo" means, so I can't go ahead.

Can you tell me what that means?

**Comment by robinkwilson** [comment, marked as answer]
I'm not sure the Reticulum repo can be run locally for outside Hubs Devs because it uses secret keys for the developer dialog (secrets configured in the hubs secret repo, sorry we can't provide access to it... It's secret!).

Another user tried running reticulum locally in this discussion: https://github.com/mozilla/hubs/discussions/3082

**Reply by robinkwilson** [reply]
Alternatively, instead of running reticulum + using the **dev** dialog servers. 

You technically could try running [dialog](https://github.com/mozilla/dialog) locally as well. You'll need to generate a secret yourself to configure the two.

**Reply by Tsuyoshina** [reply]
Now I know what "secret" means.
I'll try running dialog locally.

Thank you.

**Comment by burakakca** [comment]
You can use [here to](https://travistidwell.com/jsencrypt/demo/) for generate rsa key

I ignore the ./script/run.sh in reticulum . Run iex -S mix phx.server

Then;
- In reticulum config/dev.exs
```exs
config :ret, Ret.PermsToken, perms_key: "-----BEGIN RSA PRIVATE KEY----- paste here copyed key but add every line \n before END RSA PRIVATE KEY-----"

```
- Create `perms.pub.pem` file and add inside of `dialog/certs` then paste inside `rsa public key`

**Reply by albirrkarim** [reply]
it works, thanks

**Reply by nupur711** [reply]
> it works, thanks

How did you able to resolve, I am facing the same issue?

**Reply by albirrkarim** [reply]
with following that instruction

**Comment by albirrkarim** [comment]
I think [my article about running hubs on local and hosting it on vps](https://github.com/albirrkarim/mozilla-hubs-installation-detailed) can be helpful
